src/main/java/org/broadinstitute/hellbender/tools/coveragemodel/CopyRatioCallingMetadata.java

```diff
@@ -0,0 +1,169 @@
+        this.emissionCalculationStrategy = Utils.nonNull(emissionCalculationStrategy,
+                "Emission calculation model must be non-null");
+    }
+
+    public String getSampleName() {
+        return Utils.nonNull(sampleName, "Sample name is queried but is not set");
+    }
+
+    public Integer getSampleIndex() {
+        return Utils.nonNull(sampleIndex, "Sample index is queried but is not set");
+    }
+
+    public SexGenotypeData getSampleSexGenotypeData() {
+        return Utils.nonNull(sampleSexGenotypeData, "Sample sex genotype data is queried but is not set");
+    }
+
+    public Double getSampleCoverageDepth() {
+        return Utils.nonNull(sampleCoverageDepth, "Sample coverage depth is queried but is not set");
+    }
+
+    public EmissionCalculationStrategy getEmissionCalculationStrategy() {
+        return Utils.nonNull(emissionCalculationStrategy, "Emission calculation strategy is queried but is not set");
+    }
+
+
+    public boolean hasSampleName() {
+        return sampleName != null;
+    }
+
+    public boolean hasSampleIndex() {
+        return sampleIndex != null;
+    }
+
+    public boolean hasSampleSexGenotypeData() {
+        return sampleSexGenotypeData != null;
+    }
+
+    public boolean hasSampleCoverageDepth() {
+        return sampleCoverageDepth != null;
+    }
+
+    public boolean hasEmissionCalculationStrategy() {
+        return emissionCalculationStrategy != null;
+    }
+
+    public static SampleMetadataBuilder builder() {
+        return new SampleMetadataBuilder();
+    }
+
+
+    public static class SampleMetadataBuilder implements Serializable {
+
+        private static final long serialVersionUID = 4431390529312114201L;
+
+        final CopyRatioCallingMetadata metadata;
+
+        SampleMetadataBuilder() {
+            this.metadata = new CopyRatioCallingMetadata();
+        }
+
+        public SampleMetadataBuilder setSampleName(@Nonnull final String sampleName) {
+            metadata.setSampleName(sampleName);
+            return this;
+        }
+
+        public SampleMetadataBuilder setSampleIndex(@Nonnull final Integer sampleIndex) {
+            metadata.setSampleIndex(sampleIndex);
+            return this;
+        }
+
+        public SampleMetadataBuilder setSampleSexGenotypeData(@Nonnull final SexGenotypeData sexGenotypeData) {
+            metadata.setSampleSexGenotypeData(sexGenotypeData);
+            return this;
+        }
+
+        public SampleMetadataBuilder setSampleCoverageDepth(@Nonnull final Double sampleCoverageDepth) {
+            metadata.setSampleCoverageDepth(sampleCoverageDepth);
+            return this;
+        }
+
+        public SampleMetadataBuilder setEmissionCalculationStrategy(@Nonnull final EmissionCalculationStrategy emissionCalculationStrategy) {
+            metadata.setEmissionCalculationStrategy(emissionCalculationStrategy);
+            return this;
+        }
+
+        public CopyRatioCallingMetadata build() {
```

Comment:
@davidbenjamin The setters are private and the class is effectively immutable.
@samuelklee It is necessary: when calculating priors, we only care for sex genotypes. when calculating posteriors, we need sample read depth, emission strategy, etc.